applications/FluidDynamicsApplication/custom_elements/d_vms_dem_coupled.cpp

```diff
@@ -101,6 +102,114 @@ void DVMSDEMCoupled<TElementData>::PrintInfo(std::ostream& rOStream) const
 ///////////////////////////////////////////////////////////////////////////////////////////////////
 // Evaluation of system terms on Gauss Points
 
+template <class TElementData>
+Matrix DVMSDEMCoupled<TElementData>::GetAtCoordinate(
+    const typename TElementData::NodalTensorData& rValues,
+    const typename TElementData::ShapeFunctionsType& rN) const
+{
+    Matrix result = ZeroMatrix(Dim,Dim);
+
+    for (size_t i = 0; i < NumNodes; i++) {
+        for (size_t j = 0; j < Dim; j++) {
+            for (size_t k = 0; k < Dim; k++) {
+                result(j, k) += rN[i] * rValues[i](j, k);
+            }
+        }
+    }
+
+    return result;
+}
+
+template <class TElementData>
+double DVMSDEMCoupled<TElementData>::GetAtCoordinate(
+    const typename TElementData::NodalScalarData& rValues,
+    const typename TElementData::ShapeFunctionsType& rN) const
+{
+    return DVMS<TElementData>::GetAtCoordinate(rValues, rN);
+}
+
+template <class TElementData>
+array_1d<double, 3> DVMSDEMCoupled<TElementData>::GetAtCoordinate(
+    const typename TElementData::NodalVectorData& rValues,
+    const typename TElementData::ShapeFunctionsType& rN) const
+{
+    return DVMS<TElementData>::GetAtCoordinate(rValues, rN);
+}
+
+template <class TElementData>
+double DVMSDEMCoupled<TElementData>::GetAtCoordinate(
+    const double Value,
+    const typename TElementData::ShapeFunctionsType& rN) const
+{
+    return DVMS<TElementData>::GetAtCoordinate(Value, rN);
+}
```

Comment:
Can't you use the ones in the base `FluidElement`?

Reply:
I have tried but I can't, an alternative is to implement `GetAtCoordinate` which uses the `NodalTensorData` in the base class. I think it would be cleaner. Do you agree?

Reply:
I think this makes sense (we may need the `NodalTensorData` for other elements in the future).